cloudofficeprint/src/main/java/com/cloudofficeprint/Output/Output.java

```diff
@@ -273,6 +347,112 @@ public void setRequestOption(RequestOption RequestOption){
+     * instantiate some variables, use their default value as argument.
+     *
+     * @param filetype          This states what kind of output file type is required.
+     *                          It can be either the same as template_type ("docx",
+     *                          "pptx", "xlsx", "html", "md"), "pdf" or any other
+     *                          output file supported by libreoffice/openoffice.
+     *                          Special output type: "onepagepdf", this will cause the
+     *                          output to be converted to pdf and all the pages will
+     *                          be merged into one single page. Default : null (the
+     *                          type of the template will be used).
+     * @param encoding          This states what kind of output encoding is wished for
+     *                          the output file. It must be either "raw" (bytes) or
+     *                          "base64". Default : raw.
+     * @param converter         This states which software the server should use to
+     *                          convert the output to pdf. The Cloud Office Print
+     *                          server uses LibreOffice. If you are running the on
+     *                          premise version then the available values are :
+     *                          "officetopdf" (only when server runs on Windows ) or
+     *                          "libreoffice" (Windows, Linux, OSX)
+     *                          "libreoffice-standalone" or any other custom defined
+     *                          converters in the aop_config.json file. Default :
+     *                          libreoffice.
+     * @param appendPerPage     if you want to append file after each page of output
+     *                          set appendPerPage to true and false otherwise.
+     * @param prependPerPage    if you want to prepend file after each page of output
+     *                          set prependPerPage to true and false otherwise.
+     * @param token             If you want to store the output on a cloud based
+     *                          service, a specific CloudAccessToken object needs to
+     *                          be specified. Default : null.
+     * @param serverDirectory   If you want to save the output on the server a
+     *                          directory on the server needs to be specified. Default
+     *                          : null.
+     * @param pdfOptions        Optional PDF options. They are described in the
+     *                          PDFOptions class. Default : null.
+     * @param csvOptions        Optional CSV options. They are described in the
+     *                          CsvOptions class. Default : null.
+     */
+    public Output(String filetype, String encoding, String converter, Boolean appendPerPage, Boolean prependPerPage, CloudAccessToken token, String serverDirectory,
```

Comment:
let us not introduce new constructors, users will be able to use setters for extra parameters. 
Only Output parameter necessary is filetype.